### PR TITLE
Fix usage of pre-existing VPCs via config file

### LINKED ISSUE #436: cannot create cluster on existing VPC with config file
**What happened?**
when i created cluster on exisiting VPC with config files, it failed.

**What you expected to happen?**

**How to reproduce it?**
```
$ eksctl create cluster -f cluster.yaml
[ℹ]  using region ap-northeast-1
[ℹ]  setting availability zones to [ap-northeast-1a ap-northeast-1c ap-northeast-1d]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1d41fb8]

goroutine 1 [running]:
github.com/weaveworks/eksctl/pkg/vpc.SetSubnets(0xc00028c000, 0xc00028c000, 0x0)
        /go/src/github.com/weaveworks/eksctl/pkg/vpc/vpc.go:24 +0x148
github.com/weaveworks/eksctl/pkg/ctl/create.doCreateCluster.func3(0x3042e60, 0xc000a21500)
        /go/src/github.com/weaveworks/eksctl/pkg/ctl/create/cluster.go:301 +0x8d7
github.com/weaveworks/eksctl/pkg/ctl/create.doCreateCluster(0xc000674080, 0xc00028c000, 0x0, 0x0, 0xc0006fc780, 0x4, 0x329d123)
        /go/src/github.com/weaveworks/eksctl/pkg/ctl/create/cluster.go:361 +0x5dc
github.com/weaveworks/eksctl/pkg/ctl/create.createClusterCmd.func1(0xc0006fc780, 0xc000678ce0, 0x0, 0x2)
        /go/src/github.com/weaveworks/eksctl/pkg/ctl/create/cluster.go:53 +0x94
github.com/weaveworks/eksctl/vendor/github.com/spf13/cobra.(*Command).execute(0xc0006fc780, 0xc000678ca0, 0x2, 0x2, 0xc0006fc780, 0xc000678ca0)
        /go/src/github.com/weaveworks/eksctl/vendor/github.com/spf13/cobra/command.go:766 +0x2cc
github.com/weaveworks/eksctl/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x5154c80, 0x1, 0x5b, 0x2b)
        /go/src/github.com/weaveworks/eksctl/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/weaveworks/eksctl/vendor/github.com/spf13/cobra.(*Command).Execute(0x5154c80, 0x0, 0xc0000ac2f0)
        /go/src/github.com/weaveworks/eksctl/vendor/github.com/spf13/cobra/command.go:800 +0x2b
main.main()
        /go/src/github.com/weaveworks/eksctl/cmd/eksctl/main.go:65 +0x2d
```

- config.yaml
``` 
apiVersion: eksctl.io/v1alpha3
kind: ClusterConfig

metadata:
  name: cluster-in-existing-vpc
  region: ap-northeast-1

vpc:
  subnets:
    private:
      ap-northeast-1a: {id: subnet-099ab88b723e0xxxx}
      ap-northeast-1c: {id: subnet-0b28a3843856dxxxx}
      ap-northeast-1d: {id: subnet-0c2439e34d6a3xxxx}

nodeGroups:
  - name: ng-1-workers
    instanceType: m5.xlarge
    desiredCapacity: 10
    privateNetworking: true
```

**Anything else we need to know?**

**Versions**
```
$ eksctl version 
[ℹ]  version.Info{BuiltAt:"", GitCommit:"", GitTag:"0.1.18"}
$ uname -a 
Darwin sasaki-alpaca.local 18.2.0 Darwin Kernel Version 18.2.0: Mon Nov 12 20:24:46 PST 2018; root:xnu-4903.231.4~2/RELEASE_X86_64 x86_64
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.3", GitCommit:"2bba0127d85d5a46ab4b778548be28623b32d0b0", GitTreeState:"clean", BuildDate:"2018-05-21T09:17:39Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"darwin/amd64"}
```

**Logs**

### PR DESCRIPTION
**Pull request by errordeveloper**
### Description

- closes #436 

### Checklist
- [x] Code compiles correctly (i.e `make build`)
- [x] All tests passing (i.e. `make test`)
- [x] Added/modified documentation as required (such as the README)

### CODE REVIEW

**Comment by errordeveloper** [comment]
NOTE: there are commits from #446, that one needs merging first.

File: examples/04-existing-vpc.yaml [hunk 1]
```diff
@@ -8,19 +8,25 @@ metadata:
...
-      # NOTE: you cannot run this example directly,
-      # you will need to set actual subenet IDs here
-      eu-north-1a:
-        id: subnet-a123456
-      eu-north-1b:
-        id: subnet-b123456
-      eu-north-1c:
-        id: subnet-c123456
+    # must provide 'Private' and/or 'Public' subnets by availibility zone as shown
+    Private:
```
Comment (errordeveloper) [inline examples/04-existing-vpc.yaml]: We should make this lowercase, but that would a bigger change and we'd have to bump the API version.

**Comment by errordeveloper** [comment]
Integration test looking pass.

**Comment by pputnik** [comment]
I still have the issue. The code:
```
---
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: eksworkshop-eksctl
  region: eu-west-1

vpc:
  id: vpc-0da60ff028192627e
  subnets:
    Public: 
      eu-west-1b: { id: subnet-08dbc4f966c6410d4 }
      eu-west-1a: { id: subnet-0e0d14242f3472e8e }
      eu-west-1c: { id: subnet-02408be52718f5398 }

managedNodeGroups:
- name: nodegroup
  instanceType: t3a.mini
  desiredCapacity: 2
  iam:
    withAddonPolicies:
      albIngress: true

secretsEncryption:
  keyARN: 

```
The output:
```
[root@localhost eks]# eksctl create cluster -f eksworkshop.yaml -v4
2020-05-11T12:27:58+02:00 [ℹ]  eksctl version 0.18.0
2020-05-11T12:27:58+02:00 [ℹ]  using region eu-west-1
2020-05-11T12:27:58+02:00 [▶]  role ARN for the current session is "arn:aws:sts::172748671118:assumed-role/IzAiAdmin/1589192878185247797"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x19dcf6c]

goroutine 1 [running]:
github.com/weaveworks/eksctl/pkg/eks.validateKMSSupport(0xc000429760, 0x0, 0x1)
        github.com/weaveworks/eksctl@/pkg/eks/nodegroup.go:145 +0xac
github.com/weaveworks/eksctl/pkg/eks.ValidateFeatureCompatibility(0xc000429760, 0xc0003be3c0, 0x1, 0x1, 0xc000316b20, 0xc00050dc80)
        github.com/weaveworks/eksctl@/pkg/eks/nodegroup.go:57 +0x2f
github.com/weaveworks/eksctl/pkg/ctl/create.doCreateCluster(0xc000371b00, 0xc000348420, 0xc000371c00, 0x32ec794, 0x1ffffffffffffff)
        github.com/weaveworks/eksctl@/pkg/ctl/create/cluster.go:149 +0x5d4
github.com/weaveworks/eksctl/pkg/ctl/create.createClusterCmd.func1(0xc000371b00, 0xc000348420, 0xc000371c00, 0x0, 0x0)
        github.com/weaveworks/eksctl@/pkg/ctl/create/cluster.go:28 +0x3f
github.com/weaveworks/eksctl/pkg/ctl/create.createClusterCmdWithRunFunc.func1(0xc0003ee000, 0xc00050db90, 0x0, 0x3, 0x0, 0x0)
        github.com/weaveworks/eksctl@/pkg/ctl/create/cluster.go:43 +0xa9
github.com/spf13/cobra.(*Command).execute(0xc0003ee000, 0xc00050db60, 0x3, 0x3, 0xc0003ee000, 0xc00050db60)
        github.com/spf13/cobra@v1.0.0/command.go:842 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc0000ad340, 0xc000071ca0, 0x1, 0x1)
        github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
        github.com/weaveworks/eksctl@/cmd/eksctl/main.go:79 +0x407
```
Am I doing something wrong?

**Comment by pputnik** [comment]
I still have the issue, the vpc and subnets already created. The code:
```
---
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: eksworkshop-eksctl
  region: eu-west-1

vpc:
  id: vpc-0da60ff028192627e
  subnets:
    Public: 
      eu-west-1b: { id: subnet-08dbc4f966c6410d4 }
      eu-west-1a: { id: subnet-0e0d14242f3472e8e }
      eu-west-1c: { id: subnet-02408be52718f5398 }
    Private: 
      eu-west-1a: { id: subnet-00b9ec0511703c9f6 }

managedNodeGroups:
- name: nodegroup
  instanceType: t3a.mini
  desiredCapacity: 2
  iam:
    withAddonPolicies:
      albIngress: true
```
The error is in CF template ("Subnet Id is required"):
```
  ControlPlane:
    Type: 'AWS::EKS::Cluster'
    Properties:
      Name: eksworkshop-eksctl
      ResourcesVpcConfig:
        SecurityGroupIds:
          - !Ref ControlPlaneSecurityGroup
        SubnetIds:
          - ''
          - ''
          - ''
          - ''
          - ''
          - ''
      RoleArn: !GetAtt 
        - ServiceRole
        - Arn
      Version: '1.15'
```
Am I doing something wrong? I want my nodes be in public subnet.

**Comment by martina-if** [comment]
Hi @pputnik ,

I think there might be a mistake in the cluster configuration you are using:

```
vpc:
  subnets:
    Public: 
      eu-west-1b: { id: subnet-08dbc4f966c6410d4 }
      eu-west-1a: { id: subnet-0e0d14242f3472e8e }
      eu-west-1c: { id: subnet-02408be52718f5398 }
    Private: 
      eu-west-1a: { id: subnet-00b9ec0511703c9f6 }
```

`Public` and `Private` should be all lowercase:

```
vpc:
  subnets:
    public: 
      eu-west-1b: { id: subnet-08dbc4f966c6410d4 }
      eu-west-1a: { id: subnet-0e0d14242f3472e8e }
      eu-west-1c: { id: subnet-02408be52718f5398 }
    private: 
      eu-west-1a: { id: subnet-00b9ec0511703c9f6 }
```

Let us know if that doesn't help!

**Comment by pputnik** [comment]
Hi @martina-if , thank you, I took uppercase version it from solution in one of the linked issues.

**Pull request merged**